internal/service/infra_mgmt/api_services_data_source.go

```diff
@@ -0,0 +1,116 @@
+package infra_mgmt
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	bloxoneclient "github.com/infobloxopen/bloxone-go-client/client"
+	"github.com/infobloxopen/bloxone-go-client/infra_mgmt"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/flex"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &ServicesDataSource{}
+
+func NewServicesDataSource() datasource.DataSource {
+	return &ServicesDataSource{}
+}
+
+// ServicesDataSource defines the data source implementation.
+type ServicesDataSource struct {
+	client *bloxoneclient.APIClient
+}
+
+func (d *ServicesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "infra_services"
+}
+
+type InfraServiceModelWithFilter struct {
+	Filters    types.Map  `tfsdk:"filters"`
+	TagFilters types.Map  `tfsdk:"tag_filters"`
+	Results    types.List `tfsdk:"results"`
+}
+
+func (m *InfraServiceModelWithFilter) FlattenResults(ctx context.Context, from []infra_mgmt.InfraService, diags *diag.Diagnostics) {
+	if len(from) == 0 {
+		return
+	}
+	m.Results = flex.FlattenFrameworkListNestedBlock(ctx, from, InfraServiceAttrTypes, diags, FlattenInfraService)
+}
+
+func (d *ServicesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
+		Attributes: map[string]schema.Attribute{
+			"filters": schema.MapAttribute{
+				Description: "Filter are used to return a more specific list of results. Filters can be used to match resources by specific attributes, e.g. name. If you specify multiple filters, the results returned will have only resources that match all the specified filters.",
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+			"tag_filters": schema.MapAttribute{
+				Description: "Tag Filters are used to filter by tags return a more specific list of results. If you specify multiple filters, the results returned will have only resources that match all the specified filters.",
```

Comment:
`Tag Filters return a list of results filtered by tags. If you specify multiple filters, the results returned will have only resources that match all the specified filters.`

Would this update make sense?

Reply:
updated comment